blaze/compute/sql.py

```diff
@@ -1520,18 +1521,32 @@ def post_compute(_, s, **kwargs):
     return select(s)
 
 
-@dispatch(IsIn, ColumnElement)
-def compute_up(expr, data, **kwargs):
-    return data.in_(expr._keys)
+@dispatch(IsIn, ColumnElement, (Iterable, Selectable))
+def compute_up(expr, data, keys, **kwargs):
+    return data.in_(keys)
 
 
-@dispatch(IsIn, Selectable)
-def compute_up(expr, data, **kwargs):
+@dispatch(IsIn, ColumnElement, ColumnElement)
+def compute_up(expr, data, keys, **kwargs):
+    return data.in_(Select([keys]))
+
+
+@dispatch(IsIn, Selectable, (Iterable, Selectable))
+def compute_up(expr, data, keys, **kwargs):
+    assert len(data.columns) == 1, (
+        'only 1 column is allowed in a Select in IsIn'
+    )
+    col, = unsafe_inner_columns(data)
+    return reconstruct_select((col.in_(keys),), data)
+
+
+@dispatch(IsIn, Selectable, ColumnElement)
+def compute_up(expr, data, keys, **kwargs):
     assert len(data.columns) == 1, (
         'only 1 column is allowed in a Select in IsIn'
     )
     col, = unsafe_inner_columns(data)
-    return reconstruct_select((col.in_(expr._keys),), data)
+    return col.in_(Select([keys]))
```

Comment:
This wasn't using `select` due to unfamiliarity on my part. I'm now aware of `compute.sql.select` and I'll use that.

Going to add a test case for `expr[expr > 5]).isin([1, 2, 3]` before resubmitting.